desk/app/channels-server.hoon

```diff
@@ -655,7 +777,7 @@
         =/  post  (get:on-v-posts:c posts.channel now.bowl)
         ?~  post  now.bowl
         $(now.bowl `@da`(add now.bowl ^~((div ~s1 (bex 16)))))
-      =/  new=v-post:c  [[id ~ ~] 0 essay.c-post]
+      =/  new=v-post:c  [[id +(count.channel) id ~ ~] 0 essay.c-post]
       :-  `[%post id %set ~ new]
       (put:on-v-posts:c posts.channel id ~ new)
```

Comment:
This count gets reflected in the update that gets sent after inserting this new post. The way (as currently written) we then increment the count in channel state is that `+ca-update` looks at the update being sent and extracts the new count. This is a confusing mix of responsibilities.

I would update `+ca-c-post` to produce a modified `channel` (instead of a modified `posts.channel`), and then simply do the modification in-line here.

Shouldn't be too big a change, `+ca-c-post` only has one callsite (in `+ca-c-channel`).